tools/proofers/codepoint_validator.js

```diff
@@ -149,11 +148,24 @@ $(function () {
                 if(code) {
                     replaceChar = char1 + code;
                 } else {
-                    return;
+                    return match;
                 }
             }
             replaceChar = replaceChar.normalize("NFC");
-            maybeSubstitute();
+            return maybeReplace();
         }
-    });
+
+        text = text.replace(/\[(..)\]/gu, replacer);
+        if(validMarkup) {
+            textArea.value = text;
+            caretIndex -= displacement;
+            textArea.setSelectionRange(caretIndex, caretIndex);
+        }
+    }
+
+    // event fires on every input into the textarea
+    $(textArea).on("input", replaceMarkup);
+
+    // also replace on initial load
+    replaceMarkup();
```

Comment:
I like this!

Reply:
Perhaps we should keep this part and have the character entry work only on closing bracket.

Reply:
If it is useful to convert on load this could be done more simply. The extra complexity to put the caret in the right place would not be needed.